js/common.js

```diff
@@ -1193,3 +1193,22 @@ function relativeDate(str) {
                            : (tmp = Math.round(y)) === 1 ? __('a year ago')
                               : '%s years ago'.replace('%s', tmp);
 }
+
+/**
+ * Special case as both "English" and "English (US)" use the same locale for
+ * flatpickr but should have different first day of week.
+ * We must manually set firstDayOfWeek for "English"
+ *
+ * @param {String} locale_label
+ * @param {String} locale_code
+ * @returns
+ */
+function getFlatPickerLocale(locale_label, locale_code) {
```

Comment:
Relying on label may be a problem, why did you not pass the `$_SESSION['glpilanguage']` code only (i.e, `en_GB` or `en_US`), and then return the config or the first part of the language code matches an available locale (`typeof flatpickr.l10ns[code] !== "undefined"`), or return `default` otherwise ?

For now, we are relying on value that was used for TinyMCE, but I guess TinyMCE and Flatpickr may have different locales.

Reply:
I agree, it's not possible to rely on a label; I wonder if we should find that information from php i18n mecanisms for example

Reply:
Found `Locale::parseLocale()` which seems to do the trick.
Would be better to have something client side but I couldn't find any vanilla js functions that does this.

Reply:
Well, I had in mind something like https://www.php.net/manual/fr/intlcalendar.getfirstdayofweek.php (part of already required intl extension). BTW, first day of week is monday for fr_FR, es_ES, ... and en_GB; and sunday for en_US (among other maybe, I've tested all locales).

Reply:
@trasher 
Flatpickr already provide locales configuration that are correct in most cases (see https://github.com/flatpickr/flatpickr/tree/master/src/l10n). Problem is that locales are based only on language, and not region.

Redefining only the `firstDayOfWeek` for `en` is ok, as other values are relying on default locale which is english, but if we want to redefine this value for another locale, we will have to redefine all locale variables (see https://github.com/flatpickr/flatpickr/blob/master/src/l10n/fr.ts ).

The better way to do would be to replace inclusion of the Flatpickr locale file (`public/lib/flatpickr/l10n/xx.js`) by a call to a PHP endpoint that use PHPIntl to fill all values, taking care of the region. Such an endpoint should provide cache related headers as its content would not be subject to change in the time.

Reply:
But it is not so simple and requires new locales for Flatpick strings that are not directly related to dates.